src/mysql.c

```diff
@@ -238,6 +240,13 @@ static int mysql_config (oconfig_item_t *ci) /* {{{ */
 
 static MYSQL *getconnection (mysql_database_t *db)
 {
+	// This is necessary because we might be running on a different
+	// thread than before.
+	static pthread_mutex_t mutex = PTHREAD_MUTEX_INITIALIZER;
+	pthread_mutex_lock(&mutex);
+	mysql_library_init(0, NULL, NULL);
```

Comment:
This feels ugly.  Looking at the docs, the right thing to do is to call it before threads are spawned.  So you could just add an _init function to the plugin that you register with plugin_register_init() and then call mysql_library_init() from within that function